src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -156,87 +56,71 @@ public override int Order
             {
                 throw new ArgumentNullException(nameof(output));
             }
+            
+            IHtmlContent content = null;
 
-            IHtmlContent result = null;
             if (Enabled)
             {
                 var key = GenerateKey(context);
-                if (!MemoryCache.TryGetValue(key, out result))
+                MemoryCacheEntryOptions options;
+    
+                while(content == null)
                 {
-                    // Create an entry link scope and flow it so that any tokens related to the cache entries
-                    // created within this scope get copied to this scope.
-                    using (var link = MemoryCache.CreateLinkingScope())
+                    Task<IHtmlContent> result = null;
+                    
+                    if (!MemoryCache.TryGetValue(key, out result))
```

Comment:
Note: If you're trying to protect against multi-threading this will race. In the end all the caches will have a single value but they may have executed the body X number of times.